compiler/rustc_parse/src/parser/item.rs

```diff
@@ -65,16 +65,23 @@ impl<'a> Parser<'a> {
         }
 
         if !self.eat(term) {
-            let token_str = super::token_descr(&self.token);
             if !self.maybe_consume_incorrect_semicolon(&items) {
-                let msg = &format!("expected item, found {token_str}");
-                let mut err = self.struct_span_err(self.token.span, msg);
-                let label = if self.is_kw_followed_by_ident(kw::Let) {
-                    "consider using `const` or `static` instead of `let` for global variables"
-                } else {
-                    "expected item"
-                };
-                err.span_label(self.token.span, label);
+                if self.is_kw_followed_by_ident(kw::Let) {
+                    let mut err = self
+                        .struct_span_err(self.token.span, "`let` is invalid outside of a function");
+                    err.span_suggestion(
+                        self.token.span,
+                        "consider using `static` instead of `let`",
+                        "static",
+                        Applicability::MaybeIncorrect,
+                    );
```

Comment:
Can you use `span_suggestions` to suggest both `const` or `static`? Also, the help message on the next line is redundant, I think.

Reply:
The method is called [`span_suggestions`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/diagnostic/struct.Diagnostic.html#method.span_suggestions), not using two calls to `span_suggestion`